lxd/container.go

```diff
@@ -660,6 +679,29 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 		}
 	}
 
+	// Check that there are no contradicting root disk devices.
+	var profileRootDiskDevices []string
+	for _, pName := range args.Profiles {
+		_, p, err := dbProfileGet(d.db, pName)
+		if err != nil {
+			return nil, fmt.Errorf("Could not load profile '%s'.", pName)
+		}
+
+		k, v := containerGetRootDiskDevice(p.Devices)
+		if k != "" && !shared.StringInSlice(v["pool"], profileRootDiskDevices) {
+			profileRootDiskDevices = append(profileRootDiskDevices, v["pool"])
+		}
+	}
+
+	_, newLocalRootDiskDevice := containerGetRootDiskDevice(args.Devices)
+	if newLocalRootDiskDevice["pool"] == "" {
+		if len(profileRootDiskDevices) == 0 {
+			return nil, fmt.Errorf("Container relies on profile's root disk device but none was found")
+		} else if len(profileRootDiskDevices) > 1 {
+			return nil, fmt.Errorf("Container relies on profile's root disk device but conflicting devices were found")
+		}
+	}
```

Comment:
This part looks a bit wrong to me.

If I read it right, the following cases are considered valid even though they shouldn't:
 1) Container has a root device called "root" without a "pool" property set
 2) Container has a root device called "root" with a "pool" property set and also inherits from profile1 with a root device called "root1" and from profile2 with a root device called "root2"